docs/sql-ref-syntax-dml-insert-into.md

```diff
@@ -41,7 +41,7 @@ INSERT INTO [ TABLE ] table_identifier [ partition_spec ]
 * **partition_spec**
 
     An optional parameter that specifies a comma separated list of key and value pairs
-    for partitions.
+    for partitions. Here we support use type constructed value as partition col value.
```

Comment:
`Note that one can use a type constructed value (e.g., date'2019-01-02') for a partition column value.`?

Reply:
btw, `a type constructed value` is a common word in the user's doc? I've checked the literal doc, but I couldn't find it: https://spark.apache.org/docs/3.0.1/sql-ref-literals.html

Reply:
> Note that one can use a type constructed value (e.g., date'2019-01-02') for a partition column value.

Done

Reply:
> btw, `a type constructed value` is a common word in the user's doc? I've checked the literal doc, but I couldn't find it: https://spark.apache.org/docs/3.0.1/sql-ref-literals.html

I. will raise a pr to add this part.

Reply:
Use  `typed literal` to keep  same with 
```
  /**
   * Create a typed Literal expression. A typed literal has the following SQL syntax:
   * {{{
   *   [TYPE] '[VALUE]'
   * }}}
   * Currently Date, Timestamp, Interval and Binary typed literals are supported.
   */
  override def visitTypeConstructor(ctx: TypeConstructorContext): Literal = withOrigin(ctx) {
 
```

Reply:
Ah, I see. Do you know what's called in Hive?